File: frontend/src/api/creatorMemory.ts
import { api } from "./client";

export interface PreferenceEvidence {
  total: number;
  kept: number;
  skipped: number;
  keep_ratio: number | null;
}

export interface CreatorMemory {
  id: string;
  user_id: string;
  evidence_count: number;
  plan_feedback_count: number;
  constrained_edit_count: number;
  plan_acceptance_rate: number | null;
  optional_operation_preferences: Record<string, PreferenceEvidence>;
  preferences: Record<string, any>;
  confidence: Record<string, number>;
  summary: string;
  created_at: string;
  updated_at: string;
}

export const creatorMemoryApi = {
  get: () => api<CreatorMemory>("/users/me/creator-memory"),
  refresh: () =>
    api<CreatorMemory>("/users/me/creator-memory/refresh", { method: "POST" }),
};

File: backend/models/creator_memory.py
"""Schemas for learned creator editing preferences."""
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class PreferenceEvidence(BaseModel):
    total: int = Field(default=0, ge=0)
    kept: int = Field(default=0, ge=0)
    skipped: int = Field(default=0, ge=0)
    keep_ratio: float | None = Field(default=None, ge=0, le=1)


class CreatorMemoryOut(BaseModel):
    id: str
    user_id: str
    evidence_count: int = Field(default=0, ge=0)
    plan_feedback_count: int = Field(default=0, ge=0)
    constrained_edit_count: int = Field(default=0, ge=0)
    plan_acceptance_rate: float | None = Field(default=None, ge=0, le=1)
    optional_operation_preferences: dict[str, PreferenceEvidence] = Field(default_factory=dict)
    preferences: dict[str, Any] = Field(default_factory=dict)
    confidence: dict[str, float] = Field(default_factory=dict)
    summary: str
    created_at: datetime
    updated_at: datetime
